nodejs/eks/addon.ts

```diff
@@ -55,3 +56,39 @@ export class Addon extends pulumi.ComponentResource {
         this.addon = addon;
     }
 }
+
+export type ConfigurationValue = string | number | boolean | undefined | ConfigurationValues;
+export type ConfigurationValues = { [key: string]: pulumi.Input<ConfigurationValue> };
+
+// Sorts the keys of an object to ensure the output is deterministic.
+function stringifyReplacer(key: string, value: any) {
+    return value instanceof Object &&
+        !(value instanceof Array || value instanceof Date || value instanceof Function)
+        ? Object.keys(value)
+              .sort()
+              .reduce((sorted, key) => {
+                  sorted[key] = value[key];
+                  return sorted;
+              }, {} as { [key: string]: any })
+        : value;
+}
+
+/**
+ * Deeply merges the provided configuration values with the optional base settings and json stringifies it. Returns undefined if no values and base settings are provided.
+ * If a key exists in both the base settings and the provided values, the value from the provided values will be used.
+ */
+export function createAddonConfiguration(
+    values: pulumi.Input<object> | undefined,
+    baseSettings?: ConfigurationValues,
+): pulumi.Output<string> | undefined {
+    if (!values && !baseSettings) {
+        return undefined;
+    }
+
+    const merged = pulumi.all([values, baseSettings]).apply(([values, base]) => {
+        return deepmerge(base ?? {}, values ?? {});
```

Comment:
Based on the discussion this morning, are you going to rewrite this as part of this pr or a separate PR?

Reply:
I'm gonna remove deepmerge as part of this PR. Otherwise I won't get the build to pass :(